Observer/SalesQuoteSaveAfter.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Klaviyo\Reclaim\Helper\Data;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Plugin\Api\CartSearchRepository;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Checkout\Model\Session;
+
+class SalesQuoteSaveAfter implements ObserverInterface
+{
+    /**
+     * Klaviyo Data Helper
+     * @var Data $_dataHelper
+     */
+    protected $_dataHelper;
+
+    /**
+     * Klaviyo Scope setting Helper
+     * @var ScopeSetting $_scopeSetting
+     */
+    protected $_scopeSetting;
+
+    /**
+     * Klaviyo Cart Search Interface
+     * @var  CartSearchRepository $_cartSearchRepository
+     */
+    protected $_cartSearchRepository;
+
+    /**
+     * Magento Checkout Session
+     * @var Session $_checkoutsession
+     */
+    protected $_checkoutsession;
+
+    /**
+     * SalesQuoteSaveAfter constructor.
+     * @param Data $dataHelper
+     * @param ScopeSetting $scopeSetting
+     * @param CartSearchRepository $cartSearchRepository
+     * @param Session $checkoutsession
+     */
+    public function __construct(
+        Data $dataHelper,
+        ScopeSetting $scopeSetting,
+        CartSearchRepository $cartSearchRepository,
+        Session $checkoutsession
+    )
+    {
+        $this->_dataHelper = $dataHelper;
+        $this->_scopeSetting = $scopeSetting;
+        $this->_cartSearchRepository = $cartSearchRepository;
+        $this->_checkoutsession = $checkoutsession;
+    }
+
+    public function execute( Observer $observer )
+    {
```

Comment:
is there a reason we can't just package the data in this observer rather than needing two different ones? it seems like we should have everything needed in this observer to fire it? This has a quote, which allows us to get a masked id. it also provides us with the data to get `getLastAddedItem` from the quote and all of the items that exist in the cart?

Reply:
addressed this in the message_queues PR, the quoteId doesn't get set in the SalesQuoteProductSaveAfter event which requires the other observer. 

The other observer will save the quote Id and one of the cron jobs will get the masked Quote ID before moving to the sync table.